src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4069,26 +4069,6 @@ internal static void AddEventSource(EventSource newEventSource)
             }
         }
 
-        // Whenver we have async callbacks from native code, there is an ugly issue where
-        // during .NET shutdown native code could be calling the callback, but the CLR
-        // has already prohibited callbacks to managed code in the appdomain, causing the CLR 
-        // to throw a COMPLUS_BOOT_EXCEPTION.   The guideline we give is that you must unregister
-        // such callbacks on process shutdown or appdomain so that unmanaged code will never 
-        // do this.  This is what this callback is for.  
-        // See bug 724140 for more
-        private static void DisposeOnShutdown(object sender, EventArgs e)
```

Comment:
This should remain.  It gets used when this code is mirrored into CoreFX and is packaged into Microsoft.Diagnostics.Tracing.EventSource.Redist.

Reply:
@brianrob Thank you for the review. I modified the PR to keep DisposeOnShutdown and will make sure ILLink keeps it as well.

Reply:
Thanks much.  I don't see it here, but System.Diagnostics.Tracing.EventPipe* should also remain as well - it is currently invoked by private reflection.  Does ILLink already have a rule to keep this as well?

Reply:
SPC has InternalsVisibleTo attribute so ILLink considers all internal entry points (as well as public ones) as roots.

Reply:
Ok, thanks @erozenfeld.